src/components/user.js

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+
+import Address from './address.js';
+import Company from './company.js';
+
+export default class User extends React.Component {
+    constructor (props) {
+        super(props);
+
+        this.state = {
+            showAddress: false,
+            showCompany: false,
+            isNewUser:   this.props.isUser
```

Comment:
We don't need to put `isNewUser` in the `state` at all, because this component does not control it.
We don't use `setState` for `isNewUser`